tests/hpc/spack_master.pm

```diff
@@ -0,0 +1,82 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Summary: HPC_Module: Test Spack package installation and features
+#
+# Acts as the master node of the cluster to execute parallel executions
+#
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use Mojo::Base qw(hpcbase hpc::utils), -signatures;
+use testapi;
+use utils;
+use lockapi;
+
+sub run ($self) {
+    my $mpi = $self->get_mpi();
+    my ($mpi_compiler, $mpi_c) = $self->get_mpi_src();
+    my $mpi_bin = 'mpi_bin';
+    my @cluster_nodes = $self->cluster_names();
+    my $cluster_nodes = join(',', @cluster_nodes);
+    $self->prepare_spack_env($mpi);
+    record_info 'spack', script_output 'zypper -q info spack';
+    record_info 'boost spec', script_output 'spack spec boost';
+    assert_script_run "spack install boost+mpi^$mpi", timeout => 3600;
+    assert_script_run 'spack load boost';
+    record_info 'boost info', script_output 'spack info boost';
+
+    barrier_wait('CLUSTER_PROVISIONED');
+    ## all nodes should be able to ssh to each other, as MPIs requires so
+    $self->generate_and_distribute_ssh();
+
+    barrier_wait('MPI_SETUP_READY');
+    $self->check_nodes_availability();
+
+    record_info('INFO', script_output('cat /proc/cpuinfo'));
+
+    my $hostname = get_var('HOSTNAME', 'susetest');
+    record_info "hostname", "$hostname";
+    assert_script_run "hostnamectl status|grep $hostname";
+
+    assert_script_run("wget --quiet " . data_url("hpc/$mpi_c") . " -O /tmp/$mpi_c");
+    assert_script_run("$mpi_compiler /tmp/$mpi_c -o /tmp/$mpi_bin -l boost_mpi -I \${BOOST_ROOT}/include/ -L \${BOOST_ROOT}/lib 2>&1 > /tmp/make.out");
+
+    ## distribute the binary
+    foreach (@cluster_nodes) {
+        assert_script_run("scp -o StrictHostKeyChecking=no /tmp/$mpi_bin root\@$_\:/tmp/$mpi_bin");
```

Comment:
While this should work, I'm not a great fan of this. The directory where this is located should really be shared.

Reply:
i am planning to implement this in the near future. Which will simplify things for the compute nodes too as we discussed